pkg/backend/openshiftcluster.go

```diff
@@ -221,6 +221,11 @@ func (ocb *openShiftClusterBackend) updateAsyncOperation(ctx context.Context, lo
 }
 
 func (ocb *openShiftClusterBackend) endLease(ctx context.Context, log *logrus.Entry, stop func(), doc *api.OpenShiftClusterDocument, provisioningState api.ProvisioningState, backendErr error) error {
+	// If lease is being ended because of a backend error, log that error to user so they can troubleshoot
+	if backendErr != nil {
+		log.Error(backendErr)
```

Comment:
Not sure if this is the best place or way to log the error.

Reply:
nope.  I'm expecting to see hasResourceQuotaExceededError removed and deployARMTemplate return an *api.CloudError somehow wrapping any otherwise unhandled error returned by i.deployments.CreateOrUpdateAndWait